pir/pir-internal/src/main/java/com/duckduckgo/pir/internal/optout/PirForegroundOptOutService.kt

```diff
@@ -75,27 +70,23 @@ class PirForegroundOptOutService : Service(), CoroutineScope by MainScope() {
             createNotification(getString(R.string.pirOptOutNotificationMessageInProgress))
         startForeground(1, notification)
 
-        synchronized(this) {
-            launch(serviceDispatcher) {
-                async {
-                    val brokers = intent?.getStringExtra(EXTRA_BROKER_TO_OPT_OUT)
+        launch {
+            val brokers = intent?.getStringExtra(EXTRA_BROKER_TO_OPT_OUT)
 
-                    val result = if (!brokers.isNullOrEmpty()) {
-                        pirOptOut.execute(listOf(brokers), this@PirForegroundOptOutService, this)
-                    } else {
-                        pirOptOut.executeForBrokersWithRecords(this@PirForegroundOptOutService, this)
-                    }
+            val result = if (!brokers.isNullOrEmpty()) {
+                pirOptOut.execute(listOf(brokers), this@PirForegroundOptOutService)
+            } else {
+                pirOptOut.executeForBrokersWithRecords(this@PirForegroundOptOutService)
+            }
 
-                    if (result.isSuccess) {
-                        notificationManagerCompat.checkPermissionAndNotify(
-                            applicationContext,
-                            NOTIF_ID_STATUS_COMPLETE,
-                            createNotification(getString(R.string.pirOptOutNotificationMessageComplete)),
-                        )
-                    }
-                    stopSelf()
-                }.await()
+            if (result.isSuccess) {
+                notificationManagerCompat.checkPermissionAndNotify(
+                    applicationContext,
+                    NOTIF_ID_STATUS_COMPLETE,
+                    createNotification(getString(R.string.pirOptOutNotificationMessageComplete)),
+                )
             }
+            stopSelf()
```

Comment:
Why do we stop the service here?

Reply:
Since this is done inside a coroutine, we want to make sure that the service is stop once we get  the result.